### ISSUE TITLE
Discord Captcha Not Working

### ISSUE THREAD

**Issue by HiddenToolz** [berstend/puppeteer-extra#282]
Hello ! I tried to usecaptcha solving on Discord but it does not works !
Code : https://i.imgur.com/C6r2z6p.gif
I really don't know what to do.
Video Running : https://app.blackhole.run/#TVOC0sbnjH13YQekpE1jMdY335pPhxZZ6Gz4Zoa35NRu

**Comment by HiddenToolz** [comment]
Actualy there is a code to scan iframes, but it still not detect that thingo Oo

**Comment by alexsheets** [comment]
Yeah, I used the code provided on the homepage of the repository.
`for (const frame of page.mainFrame().childFrames()) {
        // Attempt to solve any potential reCAPTCHAs in those frames
        await frame.solveRecaptchas()
    }`

**Comment by HiddenToolz** [comment]
Same issue. Let's wait for some help then

**Comment by alexsheets** [comment]
I just tested on the following website: [https://gitlab.com/users/sign_up](url)

The captcha is within an iframe like the other website but the plugin begins solving the captcha immediately. I am unsure of the difference in why it is working here rather than on discord registration as you demonstrated or on the website I attempted to solve a captcha on. Test the website also if you like, or I can quickly share the code.

**Comment by HiddenToolz** [comment]
> it is working here rather than on discord registration as you demonstrated or on the website I attempted to solve a captcha on. Test the website also if you like, or I can quickly share the code.

I really don't need the code, except if you found a way for discord :/

**Comment by berstend** [comment]
Well, can anyone add some hard data to this discussion?

e.g.
```js
for (const frame of page.mainFrame().childFrames()) { 
  console.log(frame.url())
}
```

PS: There are [also nested frames](https://help.apify.com/en/articles/2116540-scraping-iframes-with-puppeteer), conceptually `frame.solveRecaptchas()` should work there as well.

**Comment by berstend** [comment]
> Same issue. Let's wait for some help then

I really don't appreciate this attitude, you don't provide proper info for people like me to quickly reproduce the issue.

I already closed your other ticket (#283) and will do the same here.

Please provide reproducible examples or at least debug output when asking for free support on an open-source project.

**Comment by alexsheets** [comment]
![2020-08-12](https://user-images.githubusercontent.com/60636674/90079016-55413f80-dccc-11ea-8e59-a260530ea2fd.png)

I found better success by changing the URL to the endpoint in which the provider was asking for the captcha to be done.
Then I took the code out of viewing the frames and attempting to solve the captcha because it bypassed it everytime, even though the other frame urls were coming up.
So, in essence, I did not necessarily solve my problem, but found a way to get past looking viewing the particular frames and assessing them for captchas. Perhaps you can do the same for the website you are attempting to scrape.

[https://kith.com/checkpoint?return_to=https%3A%2F%2Fkith.com%2Fcart%2F19436251414656%3A1](url)

This is the url I changed into scraping, rather than going to the product itself and then getting to the recaptcha page, where it failed before to view it.
